src/items.cpp

```diff
@@ -1359,9 +1359,10 @@ const ItemType& Items::getItemType(size_t id) const
 
 const ItemType& Items::getItemIdByClientId(uint16_t spriteId) const
 {
-	auto it = reverseItemMap.find(spriteId);
-	if (it != reverseItemMap.end()) {
-		return getItemType(it->second);
+	if (spriteId >= 100) {
```

Comment:
why is this extra check added here?

Reply:
Items in .dat file starts at the id 100, its there to prevent it from calling "more complicated" code.

Reply:
The "more complicated" code is now using a vector, which you claim is more "optimized".
Previously it was a map and there was no such check. Over-optimizing maybe?

Reply:
Yeah, this check is optional. Since i'm doing something for improving performance, i did everything i saw fit for such purpuse.